### PR TITLE
Added Integration test step and Sonar

### PR DESCRIPTION
**Pull request by kirbyAlasdairAnswer**
rewrote ci pipeline to use a multi job in parallel that then can trigger the docker build
added sonar validation
added test reporting

### CODE REVIEW

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -28,10 +31,92 @@ jobs:
           global-json-file: global.json
 
       - name: Install dependencies
-        run: dotnet restore --locked-mode
```
Comment (harrystead2) [inline .github/workflows/continous-integration.yml]: Might be wrong, but do we need --locked-mode to restore the lock.json files

Comment (kirbyAlasdairAnswer) [reply .github/workflows/continous-integration.yml]: added back in both steps

**Review by harrystead2 — DISMISSED** [review]

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -33,5 +36,87 @@ jobs:
       - name: Build
         run: dotnet build -c Release --no-restore
 
-      - name: Test
-        run: dotnet test -c Release --no-restore
+  integration-test:
+    name: Build & Test
```
Comment (milkshakeuk) [inline .github/workflows/continous-integration.yml]: Can we utilise caching here too.

**Review by milkshakeuk — COMMENTED** [review]
How come the comment isn't including coverage details? 

image

File: .github/workflows/continous-integration.yml (hunk 2 above)
Comment (kirbyAlasdairAnswer) [reply .github/workflows/continous-integration.yml]: its coverage on the code in this commit, and as this commit only has a change to a yml file theres no new coverage stats, same for duplication

Comment (kirbyAlasdairAnswer) [reply .github/workflows/continous-integration.yml]: does it not cache between jobs? can add if not

Comment (milkshakeuk) [reply .github/workflows/continous-integration.yml]: wherever you have to setup dotnet you also need to add the same caching.

**Review by milkshakeuk — DISMISSED** [review]
looks good!

**Review by milkshakeuk — APPROVED** [review]

**Review by mbaltrimas — APPROVED** [review]

**Pull request merged**